pkg/api/controller.go

```diff
@@ -369,14 +370,15 @@ func (c *Controller) StartBackgroundTasks(reloadCtx context.Context) {
 
 			// Enable extensions if extension config is provided for subImageStore
 			if c.Config != nil && c.Config.Extensions != nil {
-				ext.EnableExtensions(c.Config, c.Log, storageConfig.RootDirectory)
+				ext.EnableMetricsExtension(c.Config, c.Log, storageConfig.RootDirectory)
+				ext.EnableSearchExtension(c.Config, c.Log, storageConfig.RootDirectory)
 			}
 		}
 	}
 
 	// Enable extensions if extension config is provided for storeController
 	if c.Config.Extensions != nil {
-		if c.Config.Extensions.Sync != nil && *c.Config.Extensions.Sync.Enable {
```

Comment:
Why this change?

Reply:
Because it is a double check which would reduce the code coverage. The function EnableSyncExtension contains this logic and notifies the user with meaningful logs. These external checks would otherwise give no info whatsoever of what happened.